app/server/models/user.py

```diff
@@ -461,9 +487,17 @@ def set_pin(self, supplied_pin=None, is_activated=False):
 
         self.hash_password(pin)
 
-    # TODO(ussd): change to a field once we figure out what's the deal with reset_token
     def is_resetting(self):
-        return False
+        self.clear_expired_tokens()
+        valid_tokens = []
+        for token in self.password_reset_tokens:
```

Comment:
is it still necessary to do this loop if we've cleared the expired tokens? all that remains should be valid ones right?

Reply:
could just do `self.password_reset_tokens.count() > 0` after clearing expired i think

Reply:
Changed this, thanks :+1: